explainer.md

```diff
@@ -0,0 +1,207 @@
+
+* `disable-atomic`: Disables UA transitions for atomic navigations.
+* `disable-swipe`: Disables UA transitions for swipe navigations.
+
+The value is a space seperated list of options.
+
+### Default Value
+The default value for this setting assumes that UA transitions should be an opt-out, i.e. the UA can do a visual transition for any navigation but sites can opt-out of them (likely in favour of a custom transition). Opt-out does have the compat issue of causing double transitions for sites which already have custom transitions. But it allows browser UX to be consistent for same-document vs cross-document navigations. This also aligns with the behaviour in Webkit based browsers which already ship with UA transitions for every swipe navigation.
+
+An opt-in avoids the compat risk but conservatively disables UA transition on sites which don't have custom transitions.
+
+Note: Ideally we'd want consistent behaviour across browsers here but if its not feasible, the default value can also be left to the UA.
+
+### Handling Swipes
+In the absence of a Gesture API for customizing swipes, authors can use this API to handle swipes in the following ways:
+
+1. Add `disable-swipe` which means no visual transition occurs during the gesture. The site instead does a visual transition when the navigation commits (using the `popState` or `navigate` event). This could be sub-optimal for users since they can't use the previous page's preview to decide whether the navigation should occur.
+   However, it's unlikely that authors will do this unless a custom transition when the navigation commits is strictly better than the default UA transition.
+
+2. Omit `disable-swipe` which allows the UA to do a visual transition during the gesture. The author can detect this using the API described in [Detecting UA Transition](#detecting-ua-transition).
+
+### Global vs Navigation Based
+`same-doc-ua-transition` could be a global setting that applies to all same-document navigations originating from the current Document, or configured based on the source/destination URL. This depends on whether authors have custom transitions for a subset of navigations, or whether the behavior for swipes is set different based on the destination page.
+
+The author could simply update the value of the global setting based on what's in the navigation history, but the following cases are still infeasible:
+
+* The setting needs to be different for the back and forward entries.
+* Browser UI allows the user to traverse multiple entries, so back/fwd can jump to any entry in the navigation history.
+* The user navigates to a new URL by pasting in the URL bar.
+
+The proposal is to specify this using pair of URLs, which are the source/destination URLs for the navigation. This would effectively be a list of dictionaries which is 1:1 with a Document as follows:
+
+```
+[{
```

Comment:
The thing we briefly talked about was having something like
```
{
   from urlPattern("/articles/*") {
      to urlPattern("/index") {
           same-doc-ua-transition: disable-swipe disable-atomic;
      },
      to urlPattern("/another_index") {
           same-doc-ua-transition: disable-swipe;
      }
   }
}
```
although I suspect that's overkill. In fact, I'm kind of thinking that maybe we should just have a global without to/from things, because I'm not sure how strong the use case is for that